src/common/hotbar-store.ts

```diff
@@ -163,39 +141,52 @@ export class HotbarStore extends BaseStore<HotbarStoreModel> {
   }
 
   @action
-  addToHotbar(item: CatalogEntity, cellIndex = -1) {
+  addToHotbar(item: CatalogEntity, cellIndex?: number) {
     const hotbar = this.getActive();
+    const uid = safeCall(item?.getId);
+    const name = safeCall(item?.getName);
+
+    if (typeof uid !== "string") {
+      throw new TypeError("CatalogEntity.getId() must return a string");
+    }
+
+    if (typeof name !== "string") {
+      throw new TypeError("CatalogEntity.getName() must return a string");
+    }
+
     const newItem = { entity: {
-      uid: item.metadata.uid,
```

Comment:
Not all entities currently have `getItem()` method, so `safeCall` won't help here. And that's why @chenhunghan has an error when trying to pin some of the Links to hotbar.

Reply:
Actually, **none** of the entities can be pinned to Hotbar. 🐞